languages/csharp/exercises/concept/user-defined-exceptions/.meta/Example.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+
+public class CalculationException : Exception
+{
+    public CalculationException(int operand1, int operand2, string message, Exception inner) : base(message, inner)
+    // TODO: complete the definition of the constructor
+    {
+        Operand1 = operand1;
+        Operand2 = operand2;
+    }
+
+    public int Operand1 { get; }
+    public int Operand2 { get; }
+}
+
+public class CalculatorTestHarness
+{
+    private Calculator calculator;
+
+    public CalculatorTestHarness(Calculator calculator)
+    {
+        this.calculator = calculator;
+    }
+
+    public string Multiply(int x, int y)
+    {
+        try
+        {
+            TestMultiplication(x, y);
+        }
+        catch (CalculationException cex) when (cex.Operand1 < 0)
+        {
+            return "Multiply failed for a negative value. " + cex.InnerException.Message;
+        }
+        catch (CalculationException cex)
+        {
+            return "Multiply failed for a positive value. " + cex.InnerException.Message;
+        }
+
+        return "Multiply succeeded";
+    }
+
+    public void TestMultiplication(int x, int y)
```

Comment:
Maybe instead of returning `void`, return an `int`? I also feel that the implementations of `Multiply` and `TestMultiplication` are reversed. I think having the test harness' `Multiply` method be just a thin wrapper around the calculator's `Multiply` method might make more sense. 

The `TestMultiplication` could then have the additional behavior of converting the result of the multiplication to a `string`.

Reply:
I will swap `Multiply` and `TestMultiplication`.  

C) Again, the choice of void was deliberate.

Reply:
Okay. Swapping is good enough for me!